### PR TITLE
Auto-post on stage change and next step details

### PR DESCRIPTION
**Pull request by samschlicht**
**Job opp - posts to JobCreatorAllSourcePartners**
Screenshot 2024-05-22 at 8 45 05 AM

Screenshot 2024-05-22 at 8 45 26 AM

**Candidate opp - post to JobCreatorSourcePartner**
Screenshot 2024-05-22 at 8 48 18 AM

Screenshot 2024-05-22 at 8 49 54 AM

### CODE REVIEW

File: server/src/main/java/org/tctalent/server/service/db/impl/CandidateOpportunityServiceImpl.java [hunk 1]
```diff
@@ -104,7 +105,7 @@ public class CandidateOpportunityServiceImpl implements CandidateOpportunityServ
     private final AuthService authService;
     private final GoogleDriveConfig googleDriveConfig;
     private final FileSystemService fileSystemService;
-    private final ChatPostServiceImpl chatPostServiceImpl;
+    private final ChatPostService chatPostService;
```
Comment (sadatmalik) [inline server/src/main/java/org/tctalent/server/service/db/impl/CandidateOpportunityServiceImpl.java:108]: Good catch to use the interface here.

File: server/src/main/java/org/tctalent/server/service/db/impl/CandidateOpportunityServiceImpl.java [hunk 2]
```diff
@@ -616,9 +650,68 @@ private CandidateOpportunity updateCandidateOpportunity(
...
+                }
+            }
+
             opp.setNextStep(processedNextStep);
 
-            opp.setNextStepDueDate(oppParams.getNextStepDueDate());
+            // A next step always needs a due date
+            final LocalDate requestDueDate = oppParams.getNextStepDueDate();
+            if (requestDueDate != null) {
+                opp.setNextStepDueDate(oppParams.getNextStepDueDate());
```
Comment (sadatmalik) [inline server/src/main/java/org/tctalent/server/service/db/impl/CandidateOpportunityServiceImpl.java]:
```suggestion
                opp.setNextStepDueDate(requestDueDate);
```

**Review by sadatmalik — DISMISSED** [review]
Approved. 

Left one minor suggestion in CandidateOpportunityServiceImpl.

**Comment by samschlicht** [comment]
Thanks, @sadatmalik, good spot — I've added your small suggestion as an additional commit.

**Review by sadatmalik — APPROVED** [review]
Approved.

**Pull request merged**